### ISSUE TITLE
InterfaceTypeDefinition should contain "possibleTypes" as returned by introspection

### ISSUE THREAD

**Issue by cooniur** [graphql-java/graphql-java#2166]
Library version: v16.1

**Describe the bug**
TL;DR
For an interface type, the `possibleTypes` should be captured in `InterfaceTypeDefinition` by `IntrospectionResultToSchema`.

Details:
When introspecting an interface type, the introspection result would look like (_example taken from [this test](https://github.com/graphql-java/graphql-java/blob/master/src/test/groovy/graphql/introspection/IntrospectionResultToSchemaTest.groovy#L105)_): 

```
{
  "kind": "INTERFACE",
  "name": "Character",
  "description": "A character in the Star Wars Trilogy",
  "fields": [
    {
      "name": "id",
      "description": "The id of the character.",
      "args": [],
      "type": {
        "kind": "NON_NULL",
        "name": null,
        "ofType": {
          "kind": "SCALAR",
          "name": "String",
          "ofType": null
        }
      },
      "isDeprecated": false,
      "deprecationReason": null
    },
    ...
  ],
  "inputFields": null,
  "interfaces": null,
  "enumValues": null,
  "possibleTypes": [
    {
      "kind": "OBJECT",
      "name": "Human",
      "ofType": null
    },
    {
      "kind": "OBJECT",
      "name": "Droid",
      "ofType": null
    }
  ]
}
```

The `IntrospectionResultToSchema#createSchemaDefinition` can parse the introspection result and returns the `InterfaceTypeDefinition` object. However, the `possibleTypes` are lost during the parsing. In fact, the `InterfaceTypeDefinition` class itself does not contain a data structure to capture possible types at all (as compared to the `UnionTypeDefinition` which contains `memberTypes` that is used to contain possible types).

This would make it difficult to analyze GraphQL types programmatically because of the missing possible types information.

**To Reproduce**
Run [`IntrospectionResultToSchemaTest`](https://github.com/graphql-java/graphql-java/blob/master/src/test/groovy/graphql/introspection/IntrospectionResultToSchemaTest.groovy#L105) and look at `interfaceTypeDefinition` variable, it cannot return a list of possible types ([`Human`, `Droid`]).

**Comment by dondonz** [comment]
Hello as this issue is quite old and hasn't received any extra comments lately, I'll close this issue out.

If you'd like to make a new comment please do open a new issue, thanks!
